Src/BiometryService/BiometryService.Android.cs

```diff
@@ -0,0 +1,337 @@
+
+		private readonly CoreDispatcher _dispatcher;
+		private readonly AsyncLock _asyncLock = new AsyncLock();
+		private TaskCompletionSource<BiometricPrompt.AuthenticationResult> _authenticationCompletionSource;
+
+		public BiometryService(
+			FragmentActivity fragmentActivity,
+			Context applicationContext,
+			CoreDispatcher dispatcher,
+			FuncAsync<BiometricPrompt.PromptInfo> promptInfoBuilder)
+		{
+		
+			fragmentActivity.Validation().NotNull(nameof(fragmentActivity));
+			_dispatcher = dispatcher ?? throw new ArgumentNullException(nameof(dispatcher));
+			_promptInfoBuilder = promptInfoBuilder ?? throw new ArgumentNullException(nameof(promptInfoBuilder));
+
+			var executor = ContextCompat.GetMainExecutor(applicationContext);
+			var callback = new AuthenticationCallback(OnAuthenticationSucceeded, OnAuthenticationFailed, OnAuthenticationError);
+
+			_biometricPrompt = new BiometricPrompt(fragmentActivity, executor, callback);
+			_biometricManager = BiometricManager.From(applicationContext);
+
+			_keyStore = KeyStore.GetInstance(ANDROID_KEYSTORE);
+			_keyStore.Load(null);
+		}
+
+		public async Task<BiometryResult> ValidateIdentity(CancellationToken ct)
+		{
+			using (await _asyncLock.LockAsync(ct))
+			{
+				var response = await AuthenticateAndProcess(ct, CRYPTO_OBJECT_KEY_NAME);
+
+				var result = new BiometryResult();
+				return result;
+			}
+		}
+
+		public async Task<string> Decrypt(CancellationToken ct, string keyName, byte[] data)
+		{
+			if (this.Log().IsEnabled(LogLevel.Debug))
+			{
+				this.Log().Debug($"Decrypting the fingerprint for the key '{keyName}'.");
+			}
+
+			keyName.Validation().NotNullOrEmpty(nameof(keyName));
+			data.Validation().NotNull(nameof(data));
+
+			using (await _asyncLock.LockAsync(ct))
+			{
+				var iv = data.ToRangeArray(0, 16);
+				var buffer = data.ToRangeArray(16, int.MaxValue);
+
+				var crypto = BuildSymmetricCryptoObject(keyName, CIPHER_NAME, CipherMode.DecryptMode, iv);
+				var result = await AuthenticateAndProcess(ct, keyName, crypto) ?? throw new System.OperationCanceledException();
+				var decryptedData = result.CryptoObject.Cipher.DoFinal(buffer);
+
+				if (this.Log().IsEnabled(LogLevel.Information))
+				{
+					this.Log().Info($"Succcessfully decrypted the fingerprint for the key '{keyName}'.");
+				}
+
+				return Encoding.ASCII.GetString(decryptedData);
+			}
+		}
+
+		public async Task<byte[]> Encrypt(CancellationToken ct, string keyName, string value)
+		{
+			if (this.Log().IsEnabled(LogLevel.Debug))
+			{
+				this.Log().Debug($"Encrypting the fingerprint for the key '{keyName}'.");
+			}
+
+			keyName.Validation().NotNullOrEmpty(nameof(keyName));
+			value.Validation().NotNull(nameof(value));
+
+			using (await _asyncLock.LockAsync(ct))
+			{
+				var crypto = BuildSymmetricCryptoObject(keyName, CIPHER_NAME, CipherMode.EncryptMode);
+				var result = await AuthenticateAndProcess(ct, keyName, crypto) ?? throw new System.OperationCanceledException();
```

Comment:
in which case would `await AuthenticateAndProcess(ct, keyName, crypto)` be null and we would want to throw an exception? I would expect `AuthenticateAndProcess` to throw.